### PR TITLE
Convert familiars to GhostRoleMobSpawner

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Closes https://github.com/space-wizards/space-station-14/issues/9455. Basically, instead of spawning a familiar you now spawn a spawn point for a familiar.

:cl: Rane
- tweak: Familiars now only spawn when a ghost requests them.

### CODE REVIEW

File: Content.Server/Bible/BibleSystem.cs [hunk 1]
```diff
@@ -207,12 +222,11 @@ private void AttemptSummon(SummonableComponent component, EntityUid user, Transf
             var familiar = EntityManager.SpawnEntity(component.SpecialItemPrototype, position.Coordinates);
                             component.Summon = familiar;
 
-            /// We only want to add the familiar component to mobs
-            if (HasComp<MobStateComponent>(familiar))
+            /// If this is going to use a ghost role mob spawner, attach it to the bible.
```
Comment (metalgearsloth) [inline Content.Server/Bible/BibleSystem.cs:225]:
```suggestion
            // If this is going to use a ghost role mob spawner, attach it to the bible.
```

File: Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        /// <summary>
+        /// The entity that spawned this.
+        /// </summary>
+        public EntityUid Spawner = new();
+
+        /// <summary>
+        /// The entity spawned.
+        /// </summary>
+        public EntityUid Spawned = new();
```
Comment (mirrorcult) [inline Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs:13]:
```suggestion
        /// <summary>
        /// The entity that spawned this.
        /// </summary>
        public EntityUid Spawner;

        /// <summary>
        /// The entity spawned.
        /// </summary>
        public EntityUid Spawned;
```

File: Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+        public GhostRoleSpawnerUsedEvent(EntityUid spawner, EntityUid spawned)
+        {
+            Spawner = spawner;
+
+            Spawned = spawned;
```
Comment (mirrorcult) [inline Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs:19]:
```suggestion
            Spawner = spawner;
            Spawned = spawned;
```

File: Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+namespace Content.Server.Ghost.Roles.Events
+{
+    public sealed class GhostRoleSpawnerUsedEvent : EntityEventArgs
```
Comment (mirrorcult) [inline Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs:3]: need comments about what this event is raised on

File: Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs (hunk 3 above)
Comment (mirrorcult) [reply Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs:19]: should it even include fields at all? the spawner is presumably being deleted most of the time (so this should at least be nullable), and its being raised on the spawned entity, so

File: Content.Server/Bible/BibleSystem.cs [hunk 5]
```diff
@@ -194,6 +197,17 @@ private void OnFamiliarDeath(EntityUid uid, FamiliarComponent component, MobStat
...
 
+        /// <summary>
+        /// When the familiar spawns, set its source to the bible.
+        /// </summary>
+        private void OnSpawned(EntityUid uid, FamiliarComponent component, GhostRoleSpawnerUsedEvent args)
+        {
+            if (!TryComp<SummonableComponent>(Transform(args.Spawner).ParentUid, out var summonable))
+                return;
+
+            component.Source = (EntityUid) summonable.Owner;
```
Comment (mirrorcult) [inline Content.Server/Bible/BibleSystem.cs:208]:
```suggestion
            component.Source = summonable.Owner;
```
dont think it needs a cast?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs (hunk 3 above)
Comment (Elijahrane) [reply Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs:19]: It lets us get information about the spawner in the frame before it's deleted. It's directly used in this PR

Comment (mirrorcult) [reply Content.Server/Ghost/Roles/Events/GhostRoleSpawnerUsedEvent.cs:22]: alright, thats fine then

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**